### PR TITLE
feat: typebox

### PR DESCRIPTION
**Pull request by sinclairzx81**
This PR adds TypeBox to the benchmark test suite. This is possible as of version `0.24.0` as TypeBox now provides a specialized `TypeCompiler` that compiles it's types into efficient JavaScript validation routines.

Submitting for comparative benchmarks

### CODE REVIEW

**Comment by sinclairzx81** [comment]
@moltar Hi. Just made a few more updates to this PR. Let me know if there's anything else required to have typebox added to the test suite. Note: I've omitted the results from this PR as it's assumed these are either run on CI or your local environment, but let me know if you want me run those results from my local machine and submit them through on this PR across the node versions.

Cheers
S

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@badrap/valita": "0.1.3",
     "@marcj/marshal": "2.1.13",
     "@mojotech/json-type-validation": "3.1.0",
+    "@sinclair/typebox": "^0.24.3",
```
Comment (moltar) [inline package.json:27]:
```suggestion
    "@sinclair/typebox": "0.24.3",
```

We use fixed versions, which get upgraded automatically by Renovate Bot.

File: package-lock.json [hunk 2]
```diff
@@ -5,14 +5,14 @@
...
     "": {
-      "name": "typescript-runtime-type-benchmarks",
       "version": "1.0.0",
       "license": "MIT",
       "dependencies": {
         "@ailabs/ts-utils": "1.4.0",
         "@badrap/valita": "0.1.3",
         "@marcj/marshal": "2.1.13",
         "@mojotech/json-type-validation": "3.1.0",
+        "@sinclair/typebox": "^0.24.3",
```
Comment (moltar) [inline package-lock.json:15]:
```suggestion
        "@sinclair/typebox": "0.24.3",
```

File: package-lock.json [hunk 3]
```diff
@@ -5,14 +5,14 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "typescript-runtime-type-benchmarks",
```
Comment (moltar) [inline package-lock.json:8]: Odd that this went away?

**Review by moltar — CHANGES_REQUESTED** [review]

**Comment by moltar** [comment]
Tests on Node 10 fail, but I think we decided to get rid of Node10 anyways.

File: package-lock.json (hunk 3 above)
Comment (sinclairzx81) [reply package-lock.json:8]: @moltar I may have installed against an old version of NPM (had been running node 12 for the tests), will revert and update NPM.

**Comment by sinclairzx81** [comment]
@moltar Have decided to close off this PR, but feel free to include TypeBox in your benchmarks at a later time. I feel it's probably better for external contributors to add this benchmark rather than me personally (given the results). Feel free to use the test case from this PR if it's helpful.

Many Thanks!
S

**Comment by moltar** [comment]
@sinclairzx81 could you please elaborate why you decided not to go ahead with this PR? I am trying to understand to make sure it wasn't something with the process or something maybe I have said that offended you? Thanks.

**Comment by sinclairzx81** [comment]
@moltar Hi! No nothing like that :) I just had second thoughts on submitting benchmarks as the author of the project. I had given it some thought and feel that when taking measurements, it's generally better to have a third party to review and submit these measurements independent of the author (which would include removing the author from setting up said measurement)

I'm still quite happy to have TypeBox included, but just would prefer it was reviewed, tested, and resubmitted by someone other than me :)

Hope that clears things up!
Best Wishes
S

**Comment by moltar** [comment]
Ok, thanks for the feedback, @sinclairzx81. Appreciate it.

**Pull request closed without merging**